tests/kafkatest/tests/core/compatibility_test_new_broker_test.py

```diff
@@ -46,12 +45,12 @@ def setUp(self):
 
     @cluster(num_nodes=6)
     @parametrize(producer_version=str(LATEST_0_8_2), consumer_version=str(LATEST_0_8_2), compression_types=["none"], new_consumer=False, timestamp_type=None)
-    @parametrize(producer_version=str(LATEST_0_8_2), consumer_version=str(LATEST_0_9), compression_types=["none"], new_consumer=False, timestamp_type=None)
```

Comment:
I think the exhaustive approach is good when we can do things like label tests for infrequent/must-pass-for-final-release, but kinda wasteful nightly.

I just lost track of how we decided what to add/remove/modify from these. If we're convinced we have sufficient coverage, then I'm fine with the modifications.